source/SIL.AppBuilder.Portal.Frontend/src/data/async-waiter.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { memo, useCallback } from 'react';
```

Comment:
pulled from another project for easy intermediate state management for async behavior